protos/node.proto

```diff
@@ -0,0 +1,134 @@
+
+syntax = "proto3";
+
+import "./head.proto"
+
+package hootfs
+
+option go_package = "github.com/hootfs/hootfs/protos";
+
+// Services for working between internal nodes.
+// Initially each node will hold a cluster service connecion
+// with all other nodes in the cluster.
+//
+// This will be used for retrieving/updating 
+// information stored remotely on other nodes.
+// 
+// Initially, all information will be cloned
+// on all machines.
+// This service will take this into account.
+// i.e. data retrieval methods will not be needed
+// initially. 
+
+// ------------ MakeDirectoryCS Messages -----------------
+
+message MakeDirectoryCSRequest {
+    // Node authentication key.
+    string node_key = 1;       
+
+    // The ID of the user for which this request 
+    // originated from.
+    string user_id = 2;        
+    
+    // ID of the parent directory.
+    UUID parent_dir_id = 3;
+
+    // ID and name of the new directory.
+    UUID new_dir_id = 4;
+    string new_dir_name = 5;
+
+message MakeDirectoryCSResponse {
+    // Empty for now.
```

Comment:
We could probably add a dir_id field (or more) here to ensure that the response knows that the correct directory was made.